modules/openapi-generator/src/main/resources/go/model_inner_enum.mustache

```diff
@@ -0,0 +1,101 @@
+// {{{classname}}}{{#lambda.pascalcase}}{{{nameInCamelCase}}}{{/lambda.pascalcase}} {{{description}}}{{^description}}the model '{{{classname}}}{{#lambda.pascalcase}}{{{nameInCamelCase}}}{{/lambda.pascalcase}}'{{/description}}
+type {{{classname}}}{{#lambda.pascalcase}}{{{nameInCamelCase}}}{{/lambda.pascalcase}} {{{format}}}{{^format}}{{dataType}}{{/format}}
```

Comment:
A lot of this seems like stuff we might get for free if I could figure out how/where in `GoClientCodegen.java` to recognize that a `param.isEnum` and then create a model for that enum and update the param to reference the enum.

Reply:
please see below: https://github.com/OpenAPITools/openapi-generator/pull/16582#discussion_r1325278796